migrations/0035-add-handle-column-to-users-table.ts

```diff
@@ -0,0 +1,36 @@
+import { MigrationInterface, QueryRunner, TableColumn, TableIndex } from 'typeorm';
+
+const TABLE_NAME = 'Users';
+
+export class AddHandleColumnToUsersTable1667512340166 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    // Add "handle" column to "Users" table as nullable initially since it does not contain any data
+    await queryRunner.addColumn(TABLE_NAME, new TableColumn({
+      name: 'handle',
+      type: 'varchar(255)',
+      isNullable: true,
+      isUnique: true,
+    }));
+
+    await queryRunner.createIndex(TABLE_NAME, new TableIndex({
+      name: 'users_by_handle_index',
+      columnNames: ['handle'],
+      isUnique: true,
+    }));
+
+    // Populate the "handle" column of "Users" table in every row with a default value
+    // formatted as {firstName}-{lastName}-{hash} truncated to 32 characters
+    await queryRunner.query(`
+      UPDATE "${TABLE_NAME}"
+      SET handle = substr("firstName" || '-' || "lastName", 0, 26) || '-' || substr(md5(random()::text), 0, 7)
+    `);
```

Comment:
Is this indexing math correct? From [online](https://w3resource.com/PostgreSQL/substr-function.php) it seems like the third argument of `substr` is the number of characters to extract, which would mean that `firstName-lastName` would be 26 chars, the next dash would be 1 char, and then the final `substr` would be 7 chars, totalling 34 chars instead of 32. Have you verified that it's not 34 characters that's the limit here?